p4-14/v1.0.4/tex/p4.pt

```diff
@@ -646,11 +646,11 @@ valid bit for a metadata field, defining a separate metadata flag to
 represent that field's validity is a reasonable work around.
 \end{adjustwidth}
 
-Only valid packet header fields may result in a match (when a value is
-specified for exact or ternary matches against the field), although a
-match operation may explicitly check if a header instance (or field)
-is valid. Only valid packet headers are considered for deparsing (see
-Section~\ref{sec:deparsing}).  
+When an invalid packet header is used as a part of the match key for a table,
+the value of the header is undefined.
```

Comment:
@cc10512 It seems reasonable to me to restrict the undefined-ness to only those fields of the search key that are parts of invalid headers.  Why force the undefined-ness to spread to the whole key?  For example, if the undefined-ness is restricted to one field, and if the match_kind was ternary or lpm for that field, and every current table entry installed was "don't care" for all bits of that field, the undefined-ness doesn't make the forwarding behavior become undefined.